File: templates/python/fr/text/code/pothole.py
from karel.stanfordkarel import *
def main():
   avancer()
   remplir_les_nids_de_poule()
   avancer()

# Remplit le nid-de-poule sous la position actuelle de Karel en
# placer un jeton sur ce coin. Pour que cette fonction fonctionne
# correctement, Karel doit être face à l'est immédiatement au-dessus du
# nid de poule. Lorsque l'exécution est terminée, Karel aura
# est revenu sur la même place et sera de nouveau face à l'est.
def remplir_les_nids_de_poule():
   tourner_droite()
   avancer()
   laisser_jeton()
   tourner_autour()
   avancer()
   tourner_droite()

# Tourne Karel de 90 degrés vers la droite.
def tourner_droite():
   tourner_gauche()
   tourner_gauche()
   tourner_gauche()

# Fait tourner Karel d'environ 180 degrés.
def tourner_autour():
   tourner_gauche()
   tourner_gauche()

File: templates/java/fr/text/code/pothole.java
import stanford.karel.*;
public class NidDePoule extends Karel {
   public void run() {
      avancer();
      remplirLesNidsDePoule();
      avancer();
   }

   /**
    * Remplit le nid-de-poule sous la position actuelle
    * de Karel en plaçant un jeton sur ce coin. Pour que
    * cette méthode fonctionne correctement, Karel doit
    * faire face à l’est immédiatement au-dessus du nid
    * -de-poule. Lorsque l'exécution sera terminée, Kare
    * l sera revenu sur la même place et sera de nouveau
    * exposé à l'est.
    */
   private void remplirLesNidsDePoule() {
      tournerDroite();
      avancer();
      laisserJeton();
      faireDemiTour();
      avancer();
      tournerDroite();
   } 

   /** Tourne Karel à 90 degrés vers la droite. */
   public void tournerDroite() {
      tournerGauche();
      tournerGauche();
      tournerGauche();
   }

   /** Tourne Karel à environ 180 degrés. */
   public void faireDemiTour() {
      tournerGauche();
      tournerGauche();
   }
}